### PR TITLE
Regenerate the sitemap after a certain period of time

### PR DESCRIPTION
**Pull request by claire-lovisa**

### CODE REVIEW

**Comment by madnificent** [comment]
From a high-level perspective this looks right.  There are some things in the stack which we likely want to execute in the future which might change your approach.

There is a long-running idea that there would be a separate cron service in the triplestore which writes out events "Eg: create me a new sitemap".  Once an event comes in, a service would:

1. be notified through the delta service,
2. check the database for a new request,
3. execute its actions,
4. update the cron event so everyone knows it's been handled.

This would ensure we survive across restarts etc, and it would provide us with an administrative place to configure these common settings.  It would also give us a semi-standard trail on events which helps monitor the stack.

I'm not sure if this input changes your current approach, nor have I looked into it in sufficient depth to see if there are common mistakes in the source.  Thoughts on this high-level view are welcome and chat is fine for that too!

**Comment by erikap** [comment]
@madnificent  The sitemap is currently generated synchronously on request. But maybe I made a wrong assumption that we don't have to push the sitemap to search engines, but that they pull them at regular times?

**Comment by erikap** [comment]
@claire-lovisa Is it possible to validate the sitemap's age based on milliseconds calculations such that we don't have to include the moment library? Seems a bit like an overhead to include this library just to calculate the difference in time between 2 timestamps.

**Comment by madnificent** [comment]
@erikap It's merely something to consider.  I don't know the constraints sufficiently well to even know whether this would have an impact.  Certainly not blocking.

**Comment by claire-lovisa** [comment]
@erikap yes sure

File: app.js [hunk 1]
```diff
@@ -3,10 +3,16 @@ import { app, query } from 'mu';
 import xmlbuilder from 'xmlbuilder';
 import path from 'path';
 
-const sitemapPath = 'sitemap.xml';
+const sitemapPath = '/data/sitemap.xml';
+const sitemapMaxAge = process.env.SITEMAP_MAX_AGE || 5;
+
+if (!fs.existsSync('/data')) {
```
Comment (erikap) [inline app.js]: The folder you check here should be retrieved from the `sitemapPath` variable

**Review by nvdk — APPROVED** [review]

**Pull request merged**